storage/remote/dial_context.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"context"
+	"math/rand"
+	"net"
+	"net/http"
+
+	"github.com/prometheus/common/config"
+)
+
+type hostResolver interface {
+	LookupHost(context.Context, string) ([]string, error)
+}
+
+type customDialContext interface {
+	dialContextFn() config.DialContextFunc
+}
```

Comment:
This is only used in tests - move to _test file.

Reply:
I have removed this interface because it wasn't needed at all.